crates/aranya-daemon/src/test.rs

```diff
@@ -0,0 +1,415 @@
+                GraphId::default()
+            } else {
+                clients[0].graph_id
+            };
+            let (mut client, psk_store) = self
+                .new_client(&name, id)
+                .await
+                .with_context(|| format!("unable to create client {name}"))?;
+            stores.push(psk_store);
+            // Eww, gross.
+            if id == GraphId::default() {
+                let nonce = &mut [0u8; 16];
+                Rng.fill_bytes(nonce);
+                (client.graph_id, _) = client
+                    .create_team(DeviceKeyBundle::try_from(&client.pk)?, Some(nonce))
+                    .await?;
+            }
+            clients.push(client)
+        }
+        for store in stores {
+            let team_id = TeamId::from(*clients[0].graph_id.as_array());
+            store.insert(team_id, Arc::new(test_psk.clone()));
+            store.set_team(team_id);
+        }
+        Ok(clients)
+    }
+
+    /// Creates default team.
+    pub async fn new_team(&mut self) -> Result<Vec<TestDevice>> {
+        let mut clients = self
+            .new_group(5)
+            .await
+            .context("unable to create clients")?;
+        let team = TestTeam::new(&mut clients);
+        let owner = team.owner;
+        let admin = team.admin;
+        let operator = team.operator;
+        let membera = team.membera;
+        let memberb = team.memberb;
+
+        // team setup
+        owner
+            .actions()
+            .add_member(DeviceKeyBundle::try_from(&admin.pk)?)
+            .await
+            .context("unable to add admin member")?;
+        owner
+            .actions()
+            .assign_role(admin.pk.ident_pk.id()?, Role::Admin)
+            .await
+            .context("unable to elevate admin role")?;
+        admin.sync_expect(owner, Some(3)).await?;
+
+        let admin_caches = admin.syncer.get_peer_caches();
+        let owner_key = PeerCacheKey {
+            addr: owner.local_addr,
+            id: admin.graph_id,
+        };
+        let admin_cache_size = admin_caches
+            .lock()
+            .await
+            .get(&owner_key)
+            .unwrap()
+            .heads()
+            .len();
+        assert!(admin_cache_size > 0);
+
+        owner
+            .actions()
+            .add_member(DeviceKeyBundle::try_from(&operator.pk)?)
+            .await
+            .context("unable to add operator member")?;
+        owner
+            .actions()
+            .assign_role(operator.pk.ident_pk.id()?, Role::Operator)
+            .await
+            .context("unable to elevate operator role")?;
+        operator.sync_expect(owner, Some(5)).await?;
+
+        let operator_caches = operator.syncer.get_peer_caches();
+        let operator_cache_size = operator_caches
+            .lock()
+            .await
+            .get(&owner_key)
+            .unwrap()
+            .heads()
+            .len();
+        assert!(operator_cache_size > 0);
```

Comment:
Could hide the complexity in a test method like this:
```suggestion
        let operator_cache_size = operator.syncer.get_cache_size(&owner_key);
        assert!(operator_cache_size > 0);
```